ACF field updates

jomurgel opened the pull request:
Closes #471 

## Description
Made several general updates to improve the ACF blocks, organization, and add a couple of new features.

### Additions
Replaced the button link/button text with the ACF link field.

![screen shot 2019-03-01 at 9 15 09 am](https://user-images.githubusercontent.com/5230729/53650818-a5286580-3c02-11e9-85d4-c2495e09e7da.jpg)

I've then added a new get/display function [_s_get_link](https://github.com/WebDevStudios/wd_s/blob/hotfix/hotfix-030119-acf-field-updates/inc/acf.php#L434) which will output a link from the array and optionally a button class/additional classes.

``` php
// Defaults.
$defaults = array(
	'button' => false, // display as button?
	'class'  => '',
	'link'   => get_sub_field( 'button_link' ),
);
```

Updated the hero, carousel, and CTA blocks that are currently using a button.

Also in use is a [_s_has_array_key()](https://github.com/WebDevStudios/wd_s/blob/hotfix/hotfix-030119-acf-field-updates/inc/template-tags.php#L569) which checks to see if an array key exists, but also if the item has value.

I've also added an "overlay" toggle to the image background which has been a frequent request recently.

![screen shot 2019-03-01 at 9 18 25 am](https://user-images.githubusercontent.com/5230729/53650981-fa647700-3c02-11e9-8136-d91c4a811669.jpg)

And added that necessary info into the acf.php function: https://github.com/WebDevStudios/wd_s/blob/hotfix/hotfix-030119-acf-field-updates/inc/acf.php#L86

## Updates
Related to this was fixing some issues inside the [_hero.scss](https://github.com/WebDevStudios/wd_s/blob/hotfix/hotfix-030119-acf-field-updates/assets/sass/modules/content-blocks/_hero.scss#L6) part where the actual wrapping container and overlay wasn't, in fact, being rendered because it was expecting `.hero-block` but styling `.hero`. This also addresses a couple of z-indexing issues with the overlay.

I also removed the `$animation_class` variable from ACF template parts and instead inlined the function since they're only called once.

Opened #472 to potentially remove or update animation, but no other changes were made with that.

I removed the "title" field from the generic content block which doesn't exist.

Finally, I renamed the ACF blocks from "Group: Hero — To Clone" to "Block: Hero — To Clone" which organizes the ACF admin into blocks, global, fields, and groups. Easier to find things in my opinion.

One thing I think we COULD do, but I did not, is replace any instance of the title/text combo in blocks (CTA, Hero, Carousel, etc) with a WYSIWYG field to make the general markup and user experience better. Opened an issue to discuss this: #473

coreymcollins committed:
Checks for animation classes before trying to run animations

coreymcollins committed:
Adds animation classes to individual slides

coreymcollins committed:
Sets a transition on the slides to animate in more smoothly

coreymcollins committed:
Compiles

jomurgel committed:
updates naming convention for blocks

jomurgel committed:
updates button fields

jomurgel committed:
updates global content blocks

jomurgel committed:
replaces accidentally deleted hero field

coreymcollins committed:
Fixes an issue with carousels all using the animation of the first carousel on the page

coreymcollins committed:
Adjusts the hidden animation class

coreymcollins committed:
Compiles

jomurgel committed:
adds function to test for array key and validates value

jomurgel committed:
removes width from button field

jomurgel committed:
adds function to output class if background overlay exists (pending field)

jomurgel committed:
adds functions to get/display link from acf link array

jomurgel committed:
adds overlay background option for images

jomurgel committed:
adds hero overlay class

jomurgel committed:
removes overlay function and adds class to hero if show overlay exists

jomurgel committed:
updates styling for hero block

jomurgel committed:
cleans up hero markup — adds link function

jomurgel committed:
updates z-index of overlay in hero

jomurgel committed:
adds space befor classes in animation class function

jomurgel committed:
adds class param in link function

jomurgel committed:
adds class to hero button

jomurgel committed:
cleanup

jomurgel committed:
cleanup

jomurgel committed:
adds link/button output to carousel and cta

jomurgel committed:
removes animation variable and inlines

jomurgel committed:
builds

jomurgel committed:
Merge branch 'master' into hotfix/hotfix-030119-acf-field-updates

jomurgel committed:
adds comment

jomurgel committed:
removes title from generic content (does not exist)

coreymcollins commented on inc/acf.php:
```diff
@@ -405,3 +410,69 @@ function _s_get_theme_colors() {
+	$classes .= ' ' . $args['class'];
+
+	// Get title else default to "Read More".
+	$title = _s_has_array_key( 'title', $button_array ) ? $button_array['title'] : esc_html__( 'Read More', 'firemon' );
```
Need to update this text domain

coreymcollins commented on template-parts/content-blocks/block-cta.php:
```diff
@@ -29,8 +26,13 @@
+		_s_display_link(
+			array(
+				'button' => true,
+				'class'  => 'cta-button',
```
Can we change this to `button-cta`? We're using `button-hero` in the Hero file and I'd like to keep them consistent in naming.

coreymcollins commented on template-parts/content-blocks/block-carousel.php:
```diff
@@ -24,8 +24,6 @@
 		// Set up fields.
 		$title           = get_sub_field( 'title' );
```
Need to realign these `=` signs

coreymcollins reviewed (changes requested):
Left some code review notes with some additional bits below.

## Overlay Toggle
Can we add this to videos as well? Right now the option only exists for image backgrounds.

## Overlay Display
I have the overlay enabled, but it's not displaying. It looks like the option was added globally, but the styles were only added for the hero.

### Accordion
![](https://dl.dropbox.com/s/jllb87ybn0hzbo4/Screenshot%202019-03-01%2011.16.51.jpg?dl=0)

### CTA
![](https://dl.dropbox.com/s/plieyk74bpgalq4/Screenshot%202019-03-01%2011.02.10.jpg?dl=0)

### 50/50
![](https://dl.dropbox.com/s/gw4sgvo2035r8ji/Screenshot%202019-03-01%2011.10.02.jpg?dl=0)

### Generic Content
![](https://dl.dropbox.com/s/0y4bb4e8tsx651e/Screenshot%202019-03-01%2011.11.14.jpg?dl=0)

### Recent Posts
![](https://dl.dropbox.com/s/4wzo52ytvpu8lin/Screenshot%202019-03-01%2011.14.36.jpg?dl=0)

### Related Posts
![](https://dl.dropbox.com/s/a6g166igvlf4fyh/Screenshot%202019-03-01%2011.14.58.jpg?dl=0)

## Carousel Overlays
![](https://dl.dropbox.com/s/r8k37fo3lxd8rt3/Screenshot%202019-03-01%2011.05.35.jpg?dl=0)

I guess we've always had this, since there isn't anything new in your commits, but could be something to clean up as part of this.

There's an overlay that gets added to all of the slides in the carousel. This example is much darker than it should be. WIthout the overlay, the background would look like this:
![](https://dl.dropbox.com/s/jht5of16y5r2rf7/Screenshot%202019-03-01%2011.07.34.jpg?dl=0)

We may want to remove the fixed overlay on all slides and only add it in the case of an image or video having that toggle turned on.

## Generic Content Block
I wonder if we should keep the title block here, but have a note that it's not displayed on the frontend? The only reason I mention is because, when collapsed, most of the blocks pull the block title so you can know which block is which. It may be useful to keep the title for cases where multiple Generic Content blocks are used.
![](https://dl.dropbox.com/s/imxf4kpn79snkqx/Screenshot%202019-03-01%2011.19.10.jpg?dl=0)

jomurgel committed:
moves has-overlay styles to global content scope

jomurgel committed:
adds overlay option to video

jomurgel committed:
adds has-overlay class to videos

jomurgel committed:
removes overlay on slide by default

jomurgel committed:
removes has-overlay from inside the content block so that it will style any block

jomurgel committed:
re-adds the unused title element to acf generic content

makes note

jomurgel committed:
adds generic content note

jomurgel committed:
changes cta-button to button-cta

jomurgel committed:
updates text domain

jomurgel committed:
re-aligns =

jomurgel committed:
Merge branch 'master' into hotfix/hotfix-030119-acf-field-updates

jomurgel commented:
Made a few updates here.

I added the overlay styles to the global partial instead of the hero partial 🤦‍♀️

I also added the overlay toggle function to the video block.

![screen shot 2019-03-01 at 10 40 26 am](https://user-images.githubusercontent.com/5230729/53655782-70baa680-3c0e-11e9-8742-8019beba199e.jpg)

Which benefits from the styles, but also adds the has-overlay class to elements with the video background.

I also noticed a script error that I've resolved. This may or may not have been related to the animation class consolidation.
![screen shot 2019-03-01 at 10 44 00 am](https://user-images.githubusercontent.com/5230729/53655978-022a1880-3c0f-11e9-86d2-a0276d5afc37.jpg)

At some point the `data-animation="<?php echo esc_attr( _s_get_animation_class() ); ?>"` code was stripped from the carousel slides and the js wasn't updated, or visa versa.

I've removed the default carousel overlay and rely on the `has-overlay` class to do that work.

Made a few code tweaks:
- Realigned some `=` in the carousel.php.
- Renames the cta button to `button-cta` in the php and style parts.
- Re-added the title to the generic content blocks.
- Updated the text domain in the acf.php link function.

jomurgel committed:
resolves issue with carousel data attribute missing

jomurgel committed:
adds missing spaces in docblock

coreymcollins commented:
Just a quick note – that console error should be taken care of in the PR I submitted this morning (#471).

I'll review this again this afternoon!

coreymcollins reviewed (dismissed):
I think we're just about here on this one! With the addition of overlays to everything, it looks like there's a z-index issue with the overlay and the block content:
![](https://dl.dropbox.com/s/4po38qhj5ibir91/Screenshot%202019-03-01%2013.18.34.jpg?dl=0)

I'm unable to click on the button here and the text is sitting behind the overlay. This looks to be the case in most of the blocks, though the Related & Recent Posts and Hero block seem fine on my local.

jomurgel committed:
updates block container z-indexing

jomurgel commented:
Hmm 🤔 I thought I'd fixed that.

I've played around and addressed that issue. Looks like I did fix it, but didn't ;) That should be gtg now.

![screen shot 2019-03-01 at 1 08 03 pm](https://user-images.githubusercontent.com/5230729/53663472-1b3cc480-3c23-11e9-8ba2-43d5a03c17ac.jpg)

I added an acf-block-specific z-index to the `.container` class to ensure all of the content on all of the blocks is above the overlay at all times. Which also adds a relative position on the `.container` class as well.
- https://github.com/WebDevStudios/wd_s/pull/474/commits/47cc66f292063ffb6dd68d36ba90a6388b8b9d48

jomurgel committed:
removes inner containers and replaces wrapping elements with container class

jomurgel commented:
@coreymcollins refactored the `block-accordion.php` slightly to remove the wrapper and two internal `.container` divs to instead just have the singular `.container` wrapper. Which also resolves an overlay + background + animation issue.
- https://github.com/WebDevStudios/wd_s/pull/474/commits/ae54ba67aec88c11ddd71601f6853e2af61c329d

Tested the on the other blocks with the same setup, and as far as I can tell we're in good shape.

The only caveat in behavior is the carousel which, because of the touch-swipe feature, the text does not behave as selectable, but the z-index/overlay is correct and the button is pushable. I can't 100% determine if this is expected behavior or some unintended consequence of some style, js, or slick.

jomurgel committed:
Merge branch 'hotfix/carousel-bug' into hotfix/hotfix-030119-acf-field-updates

# Conflicts:
#	style.css
#	style.min.css
#	template-parts/content-blocks/block-carousel.php

jomurgel commented:
Going to make a few additional changes in reference to #471 to move the font/animation fields to the slides. Will add an additional update when this is complete.

jomurgel committed:
adds display options for font and animation to slide repeater — removes from parent

jomurgel committed:
updates field name for block width in carousel

jomurgel committed:
moves color and animation values into slides

jomurgel committed:
updates content blocks

jomurgel committed:
updates _s_get_animation_class output

Now accepts an array (default) and a string and checks accordingly. Updated to parse args to set that value.

jomurgel committed:
updates string logic in animation class

jomurgel commented:
I've made a couple of updates for #471 in this branch. The changes made in 471 already exists in this branch after pulling in that branch with no changes 👍 

A few things were done here:
- The class name for animation was removed from the parent carousel block (not used).
- I removed font color and animation from the parent and moved those fields to the child.
- Updated the field name for block width in the parent block and applied.
![screen shot 2019-03-01 at 2 24 53 pm](https://user-images.githubusercontent.com/5230729/53667292-d10d1080-3c2d-11e9-96bf-89758c830018.jpg)
- Updated the color class and added to the children.
![screen shot 2019-03-01 at 2 25 00 pm](https://user-images.githubusercontent.com/5230729/53667298-d5392e00-3c2d-11e9-8b53-0b3511ba48c4.jpg)
- Updated the [_s_get_animation_class()](https://github.com/WebDevStudios/wd_s/commit/82e081d236e475ae06c45cee52f0c0cfbac7c027) class which provides parsed args for the options as a param and then checks the value for either a string or an array and outputs if there's value.

Screencap of updated anmations and font color.
https://www.dropbox.com/s/c21im5fb69fel75/Kapture%202019-03-01%20at%2014.23.26.gif?dl=0

jomurgel committed:
Merge branch 'master' into hotfix/hotfix-030119-acf-field-updates

# Conflicts:
#	style.css
#	style.min.css

coreymcollins committed:
Squashed commit of the following:

commit 175671779adb2d6b4c49ded5e2987cfdb4140a08
Author: jomurgel <jomurgel@users.noreply.github.com>
Date:   Mon Apr 1 08:19:54 2019 -0600

    Hotfix/#475 phpcs updates (#476)

    * updates code standard from 1.1 to 1.2

    * resolves WDS Code phpcs 1.2.0 errors

    * resolves WDS phpcs 1.2.0 warnings

    * Fixes some additional PHPCS items from the new coding standards

    * Fixes a SassLint trailing whitespace issue

coreymcollins committed:
Compiles

coreymcollins committed:
Uses our content function for the accordion main content

coreymcollins committed:
Bumps up the z-index on the video toggle button

coreymcollins committed:
Adjusts styles for better interaction between carousel dots and video toggle

coreymcollins committed:
Replaces hero_slides with carousel_slides

coreymcollins committed:
Compiles

coreymcollins committed:
Changes the name of the background color field for better consistency

coreymcollins committed:
Adjusts the output of the block titles in the backend

coreymcollins commented:
I pushed a handful of additional updates here to take care of a few lingering things I came across:

- Replacing an instance of `wp_kses_post` with our `get_content` function
- Adjusted some z-indexing issues between carousel pagination and the video toggle 
- Replaced some instances of old field names where `hero_slides` was used instead of `carousel_slides`
- Changed the name of the background color field in the Background Options group to `background_color` instead of `color` to stay consistent with `background_image` and `background_video`
- Fixed the output of the block names in the backend

Going to pass this to Will to review as well since it encompasses a fair amount of work I've done now in addition to Jo's.

coreymcollins committed:
Some PHPCS fixes

coreymcollins committed:
PHPCS spacing fixes

ImBigWill reviewed (approved):
I think this looks good and ready for merging.

coreymcollins commented:
This was merged in manually due to the various conflicts popping up. Going to close out!
